javagen/src/main/java/com/azure/autorest/template/ModelTemplate.java

```diff
@@ -941,6 +957,18 @@ protected void writeStreamStyleSerialization(JavaClass classBlock, ClientModel m
         // No-op, meant for StreamSerializationModelTemplate.
     }
 
+    protected void addGeneratedImport(Set<String> imports) {
+        if (JavaSettings.getInstance().isDataPlaneClient()) {
+            imports.add(Generated.class.getName());
+        }
+    }
+
+    protected void addGeneratedAnnotation(JavaContext classBlock) {
+        if (JavaSettings.getInstance().isDataPlaneClient()) {
+            classBlock.annotation("Generated");
+        }
+    }
```

Comment:
If we only do this for DPG, we might instead have a `TypeSpecModelMapper` to do the override in typespec-extension.

Wait for opinion from Srikanta and Alan.

Reply:
This annotation should be extended to non-DPG and non-TypeSpec models too in the future. So, we should not make this TypeSpec specific.

Reply:
Sure.

@alzimmermsft Do you have concern if `@Generated` annotation added to vanilla (non-DPG) as well (it might affect customization jar; I assume risk is low, but not able to say no risk).

Reply:
No concerns from me with regards to the customization framework. There may be some customizations that run into minor issues but those would have likely failed due to any change in annotation generation which should be fixed anyways.